reV/SAM/econ.py

```diff
@@ -160,57 +160,79 @@ def _get_annual_energy(site, site_df, site_gids, cf_arr, inputs, calc_aey):
         return site_df
 
     @staticmethod
-    def _get_gen_profile(site, site_df, cf_file, cf_year, inputs):
-        """Get the single-site generation time series and add to inputs dict.
+    def _get_cf_profiles(sites, cf_file, cf_year):
+        """Get the multi-site capacity factor time series profiles.
 
         Parameters
         ----------
-        site : int
-            Site gid.
-        site_df : pd.DataFrame
-            Dataframe of site-specific input variables. Row index corresponds
-            to site number/gid (via df.loc not df.iloc), column labels are the
-            variable keys that will be passed forward as SAM parameters.
+        sites : list
+            List of all site GID's to get gen profiles for.
         cf_file : str
             reV generation capacity factor output file with path.
         cf_year : int | str | None
             reV generation year to calculate econ for. Looks for cf_mean_{year}
             or cf_profile_{year}. None will default to a non-year-specific cf
             dataset (cf_mean, cf_profile).
-        inputs : dict
-            Dictionary of SAM input parameters.
 
         Returns
         -------
-        inputs : dict
-            Dictionary of SAM input parameters with the generation profile
-            added.
+        profiles : np.ndarray
+            2D array (time, n_sites) of all capacity factor profiles for all
+            the requested sites.
         """
 
-        # get the system capacity
-        sys_cap = Economic._parse_sys_cap(site, inputs, site_df)
-
         # Retrieve the generation profile for single owner input
         with Outputs(cf_file) as cfh:
 
             # get the index location of the site in question
-            site_gids = list(cfh.meta['gid'])
-            isite = site_gids.index(site)
+            site_gids = list(cfh.get_meta_arr('gid'))
+            isites = [site_gids.index(s) for s in sites]
 
             # look for the cf_profile dataset
             if 'cf_profile' in cfh.datasets:
```

Comment:
What about cleaning this up a bit more to be:

```
            if 'cf_profile' in cfh.datasets:
               dset = 'cf_profile'
            elif 'cf_profile-{}'.format(cf_year) in cfh.datasets:
                dset = 'cf_profile-{}'.format(cf_year)
            elif 'cf_profile_{}'.format(cf_year) in cfh.datasets:
                dset = 'cf_profile_{}'.format(cf_year)
            else:
                msg = ('Could not find cf_profile values for '
                       'input to SingleOwner. Available datasets: {}'
                       .format(cfh.datasets))
                logger.error(msg)
                raise KeyError(msg)

            profiles = cfh[dset, :, isites]
```

Reply:
yes